comps/lvms_pg/lvm.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import json
+import os
+import time
+
+import requests
+from predictionguard import PredictionGuard
+
+from comps import (
+    LVMDoc,
+    ServiceType,
+    TextDoc,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+
+@register_microservice(
+    name="opea_service@lvm_pg",
+    service_type=ServiceType.LVM,
+    endpoint="/v1/lvm",
+    host="0.0.0.0",
+    port=8091,
+    input_datatype=LVMDoc,
+    output_datatype=TextDoc,
+)
+@register_statistics(names=["opea_service@lvm_pg"])
+async def lvm(request: LVMDoc):
+    api_key = os.getenv("PREDICTIONGUARD_API_KEY")
+    pg = PredictionGuard(api_key=api_key)
+
+    start = time.time()
+    img_b64_str = request.image
+    prompt = request.prompt
+    max_new_tokens = request.max_new_tokens
+
+    # make a request to the Prediction Guard API using the LlaVa model
+    messages = [
+        {
+            "role": "user",
+            "content": [
+                {
+                    "type": "text",
+                    "text": prompt
+                },
+                {
+                    "type": "image_url",
+                    "image_url": {
+                        "url": img_b64_str
+                    }
+                }
+            ]
+        },
+    ]
+    result = pg.chat.completions.create(
+        model="llava-1.5-7b-hf",
+        messages=messages,
+        max_tokens=max_new_tokens
```

Comment:
What if people want to modify the temperature, top_p, or top_k? Is there a way to enable this?

Reply:
I would need to add this fields to the input type. I will do that

Reply:
done